common/server_shm.hpp

```diff
@@ -25,9 +25,21 @@
 #include <boost/foreach.hpp>
 #include <boost/ref.hpp>
 #include <boost/lexical_cast.hpp>
+#ifdef _WIN32
+#    include "SC_Filesystem.hpp"
+#    define BOOST_INTERPROCESS_SHARED_DIR_FUNC
+#endif
 #include <boost/interprocess/managed_shared_memory.hpp>
 #include <boost/interprocess/containers/vector.hpp>
 
+#ifdef _WIN32
+namespace boost { namespace interprocess { namespace ipcdetail {
+inline void get_shared_dir(std::string& shared_dir) {
+    shared_dir = SC_Filesystem::instance().getDirectory(SC_Filesystem::DirName::Resource) / "shm";
```

Comment:
since `shared_dir` is a string, you have to call `.string()` on the right hand side (a `boost::filesystem::path`) to assign to it :)